### PR TITLE
Fix gitlab & bitbucket builder image detection

### PR DESCRIPTION
**Pull request by rottencandy**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5674

**Analysis / Root cause**: 
Gitlab and bitbucket detection uses `master` as the default branch when fetching resources, this fails for repositories where default branch is not `master`.

**Solution Description**: 
Avoid assuming a default branch when ref is not provided.

**Screen shots / Gifs for design review**: 

**Unit test coverage report**: 
(Unchanged)

**Test setup:**
Example repositories:
https://bitbucket.org/rottencandy/golang-ex
https://gitlab.com/rottencandy/golang-ex
https://gitlab.com/rottencandy/s2i-dotnetcore-ex (requires context dir)

**Browser conformance**: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

/kind bug

### CODE REVIEW

**Review by sahil143 — APPROVED** [review]
/lgtm

tested with mentioned repo in the description works as expected.

**Comment by invincibleJai** [comment]
verified the changes, works as expected.

File: frontend/packages/git-service/src/services/gitlab-service.ts [hunk 1]
```diff
@@ -59,13 +59,13 @@ export class GitlabService extends BaseService {
     const { name, owner, protocol, resource, full_name: fullName } = GitUrlParse(
       this.gitsource.url,
     );
-    const contextDir = this.gitsource.contextDir?.replace(/\/$/, '') || '';
+    const contextDir = this.gitsource.contextDir?.replace(/^\//, '') || '';
```
Comment (invincibleJai) [inline frontend/packages/git-service/src/services/gitlab-service.ts:62]: will good to have a test for `regex`.

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**